whep_proxy/main.go

```diff
@@ -0,0 +1,532 @@
+package main
+
```

Comment:
Could we split this file up? It's pretty large and that makes it a little hard to follow. For instance, we could move the types into it's own file, and maybe some helper functions for creating the necessary stuff to start streams.